src/chaddon.js

```diff
@@ -24,34 +30,81 @@ io.on('connection', function (socket) {
     if (addedUser) {
       return;
     };
+
     console.log("Adding: ", payload.username);
     console.log("Domain: ", payload.domain);
     // we store the username in the socket session for this client
     socket.username = payload.username;
     socket.channel = payload.domain;
     console.log("socket.channel : ", socket.channel);
 
+    //add user to channel
     socket.join(socket.channel);
 
+    if (localUser["" + socket.channel] === undefined) {
+      localUser["" + socket.channel] = {};
+    }
+    //if user isn't already in this chat somewhere else
+    if (localUser["" + socket.channel][socket.username] === undefined) {
+      localUser["" + socket.channel][socket.username] = 1;
+      //add channel to user's other tabs
+      io.in(socket.username).emit("updateRooms", {
+        room: socket.channel
+      });
+
+      socket.broadcast.to(socket.channel).emit('user joined', {
+        username: socket.username,
+        numUsers: numUsers
+      });
+    } else { //user is already in this chat
+      localUser["" + socket.channel][socket.username] = localUser["" + socket.channel][socket.username]+1;
+    }
+
     numUsers += 1;
```

Comment:
I'm not sure what you want with this. If its number of users in the room I could get the length of localUser[socket.channel] and it would likely give me that.

Reply:
The `numUsers` counter increments by 1 for every tab a user opens. 
ex If I open 4 tabs, a message will broadcast saying "4 users online"

We can open a separate issue for this.